### ISSUE TITLE
AppImage test suite based on Docker

### ISSUE THREAD

**Issue by aferrero2707** [AppImage/AppImageKit#771]
I am currently managing few AppImage packages for various image editing programs, and I have developed a set of simple scripts that allow to test an AppImage package on various Linux systems using Docker containers.

The scripts and Docker configurations can be found [here](https://github.com/aferrero2707/appimage-testsuite). The scripts basically do the following:
* build a minimal docker container using one of the supplied Dockerfile.* configurations
* allow graphical connections from the container, and start it. The scripts directory is shared under `/aitest`
* The AppImage can then be started from within the container with the following command:
```
/aitest/aitest.sh
```
which in turn extracts the package with `bsdtar` and then runs the `AppRun` executable.

The advantage is that in case of failure the user can directly modify the extracted AppImage or install additional packages.

Would you find this useful and worth being integrated into your AppImage tools?

**Comment by aferrero2707** [comment]
No, I am mostly using it to check/detect issues related with fonts or other UI elements, or to check the overall performance of the AppImage in comparison to native packages.

**Comment by probonopd** [comment]
> Would you find this useful and worth being integrated into your AppImage tools?

I think this is great but should best be maintained in its own project, not in https://github.com/AppImage/AppImageKit. Happy to link to it, though!

**Comment by azubieta** [comment]
This seems really useful! @probonopd @TheAssassin what do you think about forking this project into AppImageKit and create a reference from our doc.appimage.org so developers can know about it.

**Comment by TheAssassin** [comment]
Mentioning in the docs is a great idea. But why fork?

@aferrero2707 did you notice that we created https://docs.appimage.org? It would be a great place for a guide on your tool (as part of the Packaging Guide).

**Comment by aferrero2707** [comment]
@TheAssassin @probonopd I have added a section about my appimage testsuite in the docs, see this pull request: https://github.com/AppImage/docs.appimage.org/pull/16
